### PR TITLE
RFC Adding Derived Key Feature

### PR DESCRIPTION
**Pull request by AdithyaKrishnan**
Added Derived Key Feature set. We can now create a derived key and display it in hex format.

### CODE REVIEW

**Comment by tylerfanelli** [comment]
@AdithyaKrishnan Can you rebase and squash the commits?

File: .vscode/launch.json
Comment (larrydewey) [inline .vscode/launch.json:1]: We should add `.vscode` to the `.gitignore` file so we aren't messing with peoples configurations :)

File: src/key.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::*;
+use sev::firmware::guest::{DerivedKey, Firmware, GuestFieldSelect};
+//use std::{fs, path::PathBuf, str::FromStr};
```
Comment (larrydewey) [inline src/key.rs:5]: We can probably remove this if we are going to comment it out.

File: src/key.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+use std::{fs, path::PathBuf};
+use std::io::Read;
+
+#[derive(StructOpt)]
+pub struct KeyArgs {
+#[structopt(help = "This is the directory where the derived key will be saved")]
+pub key_path: PathBuf,
```
Comment (larrydewey) [inline src/key.rs:12]: Go ahead and run `cargo fmt` to make sure this is indented properly.

File: src/key.rs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+Ok(())
+}
+
+pub fn read_key(key_path: PathBuf) -> Result<Vec<u8>, anyhow::Error> {
+    
+    let mut key_file = fs::File::open(key_path)?;
+    let mut key = Vec::new();
+    // read the whole file
+    key_file.read_to_end(&mut key)?;
+    //let key_report = bincode::deserialize_from(key_file)
```
Comment (larrydewey) [inline src/key.rs:113]: Remove the commented code here, as well.

**Review by larrydewey — CHANGES_REQUESTED** [review]
Hey @AdithyaKrishnan here are a few other things to change, as well.

**Pull request closed without merging**